### PR TITLE
Fix inversion of pluginId and pluginType parameteres in DLQ entry creation

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

Fixes a bug in the DLQ writer, which inverted the pluginId and pluginType attributes.

## What does this PR do?

Pass the argument in correct order when calls `DeadLetterQueueWriter.writeEntry` from `PluginDeadLetterQueueWriterExt`.
Adds a unit test to verify the DLQ writer-read interaction.

## Why is it important/What is the impact to the user?

When the DLQ input receives an event now `pluginId` and `pluginType` are properly valued.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [x] run real test
- [x] cover with test to avoid regression

## How to test this PR locally

Executes the suggested steps in [original issue](https://github.com/logstash-plugins/logstash-input-dead_letter_queue/issues/48#issuecomment-1427687540).

## Related issues

- Fixes #https://github.com/logstash-plugins/logstash-input-dead_letter_queue/issues/48

### CODE REVIEW

File: logstash-core/src/test/java/org/logstash/common/AbstractDeadLetterQueueWriterExtTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    private DeadLetterQueueReader openDeadLetterQueueReader(Path dlqPath, String dlqName, String segment) throws IOException {
+        DeadLetterQueueReader dlqReader = new DeadLetterQueueReader(dlqPath);
+        Path currentSegmentFile = dlqPath.resolve(dlqName).resolve(segment);
+        dlqReader.setCurrentReaderAndPosition(currentSegmentFile, DLQ_VERSION_SIZE);
+        assertEquals(currentSegmentFile, dlqReader.getCurrentSegment());
+        return dlqReader;
+    }
+
+    private void writeAnEventIntoDLQ(Path dlqPath, String pluginId, String pluginType, String dlqName) {
```
Comment (kaisecheng) [inline logstash-core/src/test/java/org/logstash/common/AbstractDeadLetterQueueWriterExtTest.java:58]: I am wondering if other tests also write an event to DLQ and could make this method to a util class

**Review by kaisecheng — COMMENTED** [review]
Implementation looks good. Left a question about the test.

File: logstash-core/src/test/java/org/logstash/common/AbstractDeadLetterQueueWriterExtTest.java (hunk 1 above)
Comment (andsel) [reply logstash-core/src/test/java/org/logstash/common/AbstractDeadLetterQueueWriterExtTest.java:58]: `DeadLetterQueueWriterFactory` is used in its test https://github.com/elastic/logstash/blob/main/logstash-core/src/test/java/org/logstash/common/DeadLetterQueueFactoryTest.java but I haven't found test where a `AbstractDeadLetterQueueWriterExt` is created and initialized with an instance of the real DLQ writer.

**Review by kaisecheng — APPROVED** [review]
LGTM

**Comment by andsel** [comment]
@logstashmachine backport 8.7

**Pull request merged**